### DISCUSSION TITLE
Changes to primitive tools needed.

### DISCUSSION THREAD

**Discussion by BitBite112** [CleverRaven/Cataclysm-DDA#47211] [Ideas]
I made a discussion a few days ago about the stone chopper being too difficult to make, but it seems that this is not the only issue.
There has been an overhaul for the primitive tools which is neither realistic or balanced in a good way, at least the flintknapping part is. It made it so that you require a higher skill for a simpler item and a lower skill for an item that should be more difficult to make.

Not only this, but in order to make a stone adze you need a stone hammer which requires a stone chisel to make. 
Based on the materials required for the creation of a stone chisel it seems to be emulating grinding a stone into a chisel, but rather poorly. It's got the idea right, using a rock on another one with an abrasive such as sand and splashing water on it, but the problem is, is that it only allows sand and not soil. There's this guy on YouTube who shows that you don't even need sand or soil, you can simply grind another stone on a rock which makes it's own sand although it does take some extra time.

If I want to start an innawoods playthrough (I really want to since I'm not too experienced in the game's combat yet) I'll have to first try and get a clay pot which requires a fire. I could use sticks, but those don't last very long and I'd need a large amount. I would need a stone adze to get logs from tree trunks, but to get that I need all of these tools. These tools aren't bad, but they've got a somewhat high requirement, my main issue being not having a cutting tool before being able to level up fabrication and survival without having a cutting tool. 

I also said it's not realistic, one reason being that sand is required for grinding the chisel and not just soil and another being that the idea of needing a high skill to flintknap at all is just wrong. There are tiers of stone tools called modes. Each mode requires more skill to make than the previous one and can be as simple as just smashing a rock on the ground or banging two together on an anvil. You need a rather high level of fabrication and survival before being able to craft a stone chopper which should be a rather simple stone tool. I know there's a bone shiv, but that requires a very large amount of RNG from bushes or requires butchering an animal which needs... A cutting tool.

The game is currently not very playable for me since I don't want to go into the cities and towns as of yet.

**Comment by BitBite112** [comment]
Would anyone know how this could be implemented? I don't know much about coding. Do coders often look at discussions?

**Reply by actual-nh** [reply]
BTW, I am pretty sure all of this could be done via JSON alterations, not C++. (And I would guess people would vary on whether they read discussions...) You might want to take a look at the github wiki.

**Comment by ZhilkinSerg** [comment]
All of the changes in #41711 were JSON, so you can start adjusting existing (or add alternative) recipes for primitive tools and make a proposal either here or directly as pull request.

Use following guide - https://github.com/CleverRaven/Cataclysm-DDA/wiki/Guide-to-adding-new-content-to-CDDA-for-first-time-contributors.

**Comment by actual-nh** [comment]
Comments on #47550?

**Comment by Fosheze** [comment]
I'll see if I can look into some of these later this week. I needed something easy to get used to the github workflow anyways.

**Comment by kevingranade** [comment]
> There are tiers of stone tools called modes. Each mode requires more skill to make than the previous one and can be as simple as just smashing a rock on the ground or banging two together on an anvil.

Do you have any documentation you can link for this?

**Reply by actual-nh** [reply]
Very interesting! From Shea's article, one element of tool creation that I'm not sure the game handles currently is potential injury - particularly when "cold, tired, frightened, and disoriented" (or otherwise reduced in Dexterity and possibly Perception) or when using some particular techniques (e.g., throwing rocks at another, larger and hopefully suitable as a "hammerstone" rock).

Also in the article:

- Pages 11-12 have somewhat of a summary of what someone with no experience is likely to be able to do (and to some degree even think of doing). Once a bit of practice is gained, in terms of modes (see the technical summary) A-C and D4 appear the easiest.
- Tool use, including butchery, with some information on how it wears out stone tools.

**Reply by kevingranade** [reply]
https://www.academia.edu/14831665/Shea_J_J_2015_Making_and_Using_Stone_Tools_Advice_for_Learners_and_Teachers_and_Insights_for_Archaeologists_Lithic_Technology_40_231_248
Looks like a really great source.

This is a very informative chunk relative to learning time.
> How long it will take to learn to make stone tools depends what one wants to make.  Knapping flakes with non-hierarchical cores (bipolar cores, pebble cores) ought to take no more than a few minutes, and it can be done with any conchoidally-fracturing rock.  A few kilograms of such rock are sufficient for this purpose.  Learning how to hierarchically retouch flakes is also relatively easy and can be done with similar materials.  If one wants a retouched edge that is straight, use the ventral side of an edge as the striking platform surface and the dorsal side as the flake detachment surface.  Otherwise the retouched edge follows the dorsal surface's irregular contour.Learning how to make relatively simple elongated non-hierarchical cores, such as handaxes and other "long core tools" (LCTs) ought to take another hour or so.  Making LCTs requires large and flatter pieces of higher quality lithic material (large flakes or tabular pieces of rock) amounting to tens of kilograms.  More complex forms of elongated non-hierarchical cores, such as thinned bifaces, tanged bifaces/points, and celts may require several more hours of practice.  Bifacial hierarchical cores and blade cores require a bit more time (3-6 hours) and at least 10-50 kilograms of high-quality lithic raw material. 

Keeping in mind this estimate is for, "in a class with an expert knapper who is also an experienced teacher."
It provides no such estimate for experimentation-based learning, but it's safe to say this is a lower bound that shouldn't be reachable with experimentation-based learning.

Also, there is a callout for material:
> Knappable stone, i.e., stone with conchoidal fracture properties, can be found in nearly every part of the world.  The most obvious places to seek such rock in bulk are road-cuts and geological faults.  However, bedrock exposures of brittle rock often contain crystallization planes and incompletely propagated fractures that can be obstacles to fracture propagation.  Rivers, streams, gullies and beaches located downslope from bedrock sources can contain clasts (rounded rocks) and angular rock fragments with fewer internal fractures in them.

It is reasonable to restrict the supply of knappable rocks (what the game calls sharp rock), as currently finding them is utterly trivial, whereas in reality, it can be a rather major blocker to getting started.  Since we don't track geological features of any kind, the only distribution method that leaves us is to place suitable stones in streambeds, and ideally an adjustment to mapgen to place smaller streams that would be readily navigable.

There is also a nice section on using stone tools, that essentially reiterates again and again that they are much harder to use than metal tools. for pretty much every purpose.  We definitely need a wide gap between the effectiveness of these tools and modern ones.  A very critical feature is that all stone tool are inherently disposable.  They break and require either retouching or replacement on the order of minutes to days, not years as we are used to with metal tools.

Switching from summary to proposal, essentially following the modes as established.
First tier is producing sharp flakes from a core, no proficiency required, just a small flakeable stone and a hammer.  A rapidly gained proficiency would improve time to craft, and ideally yield for a given stone.
Flakes would essentially be disposable cutting implements capable of a certain amount of work before becoming unusable.  They would be suitable for many butchery tasks, but useless for harder materials such as wood, and completely useless for combat.  In addition to being the primary tool used at first, most other knapping activity would also produce these as a side effect, so they would stay readily available as long as knapping continued.
Incorporating the core itself as a usable tool gets rather complex.  On one hand, we want to represent it because you do end up with some sharp but robust edges, which would be much more useful for heavier work like breaking bones and cutting wood, but on the other hand it means we need a system for tracking when the core becomes sharp enough to use, what kind of tool qualities it has, when regular use removes the sharpness and tool qualities, and the point at which it becomes too small for use either as a stone tool or as a knapping core.
Possibly we would get away with a single tool type for pebble cores, that is marginally useful for butchery and very minor woodworking.

The second tier is a dedicated chopper, this would require the basic knapping proficiency gained from just flaking rocks, and ideally would only unlock after a certain amount of experience with both knapping and using incidentally-created pebble cores as tools, which we would probably represent as different proficiencies.  It would also require a higher quality of source stone, possibly just represented as a larger volume source stone that is distributed more rarely along with the pebble-sized stones.
The resulting tool would either be a better chopper or possibly just a more reliably-created chopper.  Ideally this is the point at which tool differentiation occurs, as is the hallmark of "Mode 2", but I'm not sure how to usefully represent that, and I'm not sure it adds anything to the game compared to simply having a progression from flakes as tools to cores as tools.

The third tier is characterized by a large jump in both sophistication required and tool effectiveness.  Another tier of proficiencies gates these tools, and probably require a still higher tier of material requirements.  The tools are still stone and subject to rapid wear, but they start being more effective at their jobs while they last, and lasting a bit longer.  The hand axe makes its appearance here, allowing the ability to take down small trees reasonably.  I'd need to take a look at what other tools are available, but I expect awls and choppers and other specialty tools.

A fourth tier would bring in weapon points, as they require not just more sophisticated shaping techniques, but also ancilliary tools such as Burins to fit components together.  I'm honestly not that interested into getting too deep into that at this point, let's get a more solid foundation in place before we worry too much about that.

Abrasion techniques are essentially orthogonal, as they don't really require the same degree of skill as knapping.

**Reply by anothersimulacrum** [reply]
> Keeping in mind this estimate is for, "in a class with an expert knapper who is also an experienced teacher."
It provides no such estimate for experimentation-based learning, but it's safe to say this is a lower bound that shouldn't be reachable with experimentation-based learning.

[Teaching to make stone tools: new experimental evidence supporting a technological hypothesis for the origins of language](https://www.nature.com/articles/s41598-017-14322-y) covers less effective teaching methods to some extent, but it is still not experimentation based.

**Reply by actual-nh** [reply]
> Abrasion techniques are essentially orthogonal, as they don't really require the same degree of skill as knapping.

Some degree of subtlety may be present in selection of materials, giving rise to different rates of progress; anything on that would require more differentiation in, e.g., hardness of stones.

**Reply by kevingranade** [reply]
Upon further thought, I still think the tiers outlined with associated "modes" is the way to go, but the outputs of that probably shouldn't be differentiated except at the absolute highest tier of knapping.  i.e.
Mode 1
Disposable flakes, ok for all light duty cutting. (Side note, macuahuitl)
Sharp cores, terrible at but capable of light and medium-duty cutting tasks.
Mode 2
(still flakes, same properties)
Chopper, kind of ok at light and medium-duty cutting tasks, terrible at but capable of some heavy-duty cutting tasks.
Mode 3
(still flakes, same properties)
Hand Axe, decent at medium duty cutting and passable at heavy duty cutting. Almost incidentally ok at combat.
Mode 4+
{flakes, flakes never change)
Hafted hand axe, incremental improvements across the board.
Spear and arrow points.
Stone knife?

Somewhere in the above progression triangular stone pieces suitable for use as wood drills appear, this is a Big Deal.

Still, across the board, wear, damage and destruction of these tools even when used correctly is an inherent property of knapped stone, they must have this property to be at all representative.  Even the best hand axe or stone knife will chip and become unusable without rework in a very short amount of time, and the amount of rework they can sustain is very limited.

**Reply by robotic-gamer** [reply]
some thoughts (sorry if this is necro I don't know whether or not it is), also sorry if this is rambly/unhelpful

-it seems like it would be useful if items health were defined as a fraction
          > the idea is the fraction defined by a top variable and bottom variable (the top can at most equal the bottom)
          >the fraction indicates the percentage of the tools usability, or health which affects its performance
          >every time the item takes damage/wear the top variable goes down
          >repairing restores the top variable to an extent but lowers the bottom variable
          >each repair restores at most a quarter
          >how close the top can be to the bottom is determined by the quality of the repair
          >the repair quality is an decimal that
                    ~determines the best fully repaired ratio
                    ~the bottom is multiplied by the repair tool each repair
                    ~ex say something 50/100 is repaired by a repair tool of 0.8  the net output is
                      50/100--->50/80----->70/80------>64/80
          >the end result is you can only repair tools so many times before it becomes impractical
-I get the feeling this implementation is a terrible idea for reasons i am oblivious to (like how the code works) hopefully the basic idea is useful

-back on to knapping isn't a really simple way to make a cutting tool just to throw a rock at an larger rock (say something with an anvil quality of 1) repeatedly until the rock breaks in a hopefully useful way.

edit: looked back at what was said and noticed actual-nh already mentioned throwing rocks

edit :also worth mentioning that finding metal or glass could happen during playthroughs resulting in them being potential materials/tools for making tools. glass could be used as a inferior version of obsidian.

**Comment by anothersimulacrum** [comment]
https://github.com/CleverRaven/Cataclysm-DDA/pull/50044
